tests/new_api/test_device.py

```diff
@@ -729,3 +726,337 @@ def test_derivatives_tape_batch(self, phi, execute_and_derivatives, batch_obs):
         assert len(jacs[0]) == len(expected_jac[0])
         assert np.allclose(jacs[0], expected_jac[0])
         assert np.allclose(jacs[1], expected_jac[1])
+
+
+@pytest.mark.parametrize("batch_obs", [True, False])
+class TestVJP:
```

Comment:
Most of these tests are copies of the `TestDerivatives` class. I _could_ have parametrized the tests in that class, but it would've made the tests a lot less readable so I decided to make a separate class with tests dedicated to VJPs